### PR TITLE
Flatlist with animation

### PR DESCRIPTION
**Pull request by zhmurko-anastasi**
https://user-images.githubusercontent.com/53611771/139914230-7aaf6d58-6a61-4f25-af34-dd5fe42d29e3.mp4

### CODE REVIEW

File: src/modules/list/ListScreen.tsx [hunk 1]
```diff
@@ -1,37 +1,72 @@
...
+import { GET_ITEMS } from './graphql';
+import { LoadingIndicator } from '../../components/statuses/LoadIndicator';
+
+type PROP = {
+  id: string,
+  title: string,
+  description: string
+}
+
+const keyExtractor = (item: { id: any }): any => {
```
Comment (Ragnarreck) [inline src/modules/list/ListScreen.tsx:20]: Types here?

File: src/modules/list/ListScreen.tsx [hunk 2]
```diff
@@ -1,37 +1,72 @@
...
+const listEmptyComponent = () => {
+  return (
+    <Text style={styles.emptyText}>You haven't any words yet</Text>
+  )
+}
+
 export function ListScreen({ navigation }: RootStackScreenProps<Routes.list>) {
+  const { data, loading, error } = useQuery(GET_ITEMS);
+
+  const renderItem = React.useCallback(({ item }: PROP | any) => {
```
Comment (Ragnarreck) [inline src/modules/list/ListScreen.tsx:33]: If you have `|any`, it's basically just `any` :)

File: src/modules/list/ListScreen.tsx [hunk 3]
```diff
@@ -1,37 +1,72 @@
...
+  }
+
   return (
-    <View style={styles.container}>
-      <Pressable onPress={() => navigation.replace(Routes.login)} style={styles.link}>
-        <Text style={styles.linkText}>List screen!</Text>
-      </Pressable>
-    </View>
+    <FlatList
+      refreshing
```
Comment (Ragnarreck) [inline src/modules/list/ListScreen.tsx:46]: `refreshing` should be changed based on `loading`

File: src/modules/list/components/ListItem.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+import { Animated, Pressable, Text, StyleSheet } from 'react-native';
+import {Colors} from '../../../constants/Colors';
+
+type PROP = {
+    id: string,
+    title: string,
+    description: string
+  }
+
+const ListItem = ({item}: PROP | any) =>{
```
Comment (Ragnarreck) [inline src/modules/list/components/ListItem.tsx:11]: Same, remove `any`

File: src/modules/list/ListScreen.tsx (hunk 2 above)
Comment (zhmurko-anastasi) [reply src/modules/list/ListScreen.tsx:33]: Yeah, it's my 'favorite' topic - types )) It just didn't want just to work without 'any; =) But I'll get it worked!

File: src/modules/list/ListScreen.tsx (hunk 3 above)
Comment (zhmurko-anastasi) [reply src/modules/list/ListScreen.tsx:46]: Ooo, certainly, will be fixed!

File: src/modules/list/ListScreen.tsx (hunk 1 above)
Comment (Ragnarreck) [reply src/modules/list/ListScreen.tsx:14]: Still any here. Can we assume it's `string`?

**Review by Ragnarreck — APPROVED** [review]
LGTM after fixing the last `any`

**Comment by AND-GORNIY** [comment]
Exciting animation, really cool. LGTM

**Pull request merged**